docs/DeveloperGuide.md

```diff
@@ -1,38 +1,254 @@
 # Developer Guide
 
-## Acknowledgements
+## Introduction
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+Budget Buddy aims to help those who want a Command Line Interface App that is a one-stop solution to keep track of their
+expenses, help them with budgeting and also be help them to visualise what they have spent.
 
-## Design & implementation
+Overspending and the lack of savings is a significant problem found in most fresh graduates. We want to create tools
+necessary for them so that they can realize their actual spending and achieve financial freedom in the future.
+
+## Setting up the project
+
+## Design
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
```

Comment:
This is useful, but maybe you can rephrase to describe it better to the reader what you are trying to explain.
e.g.
Design and implementation will be described here. UML diagrams and code snippets will be used ......